### ISSUE TITLE
False warning is shown for BatchNormalization when batch size is 1, but tensor dimensions are larger than 2

### ISSUE THREAD

**Issue by tkerola** [chainer/chainer#4662]
https://github.com/chainer/chainer/blob/master/chainer/functions/normalization/batch_normalization.py#L90-L100

The above warning is correct only when the input tensor has 2 dimensions and batch size 1. E.g. with input `x = np.random.rand(1, 3, 4, 5)`, BatchNormalization works fine, since the statistics are aggregated over the 3rd and 4th axis as well, so this warning is confusing e.g. when training large semantic segmentation models that only fit in memory with batch size 1.
